php/blocks/controls/class-checkbox.php

```diff
@@ -44,15 +44,8 @@ public function __construct() {
 	 * @return void
 	 */
 	public function register_settings() {
-		$this->settings[] = new Control_Setting(
-			array(
-				'name'     => 'help',
-				'label'    => __( 'Help Text', 'block-lab' ),
-				'type'     => 'text',
-				'default'  => '',
-				'sanitize' => 'sanitize_text_field',
-			)
-		);
+		$this->settings[] = new Control_Setting( $this->settings_config['location'] );
```

Comment:
@lukecarbis, your idea of adding the location setting only when needed was good. This does that.